pkg/cmd/render/render.go

```diff
@@ -63,6 +81,16 @@ func (r *renderOpts) Validate() error {
 		return err
 	}
 
+	// Validate all files are specified when specifying infrastructure and configmap files
+	if len(r.clusterInfrastructureInputFile) != 0 || len(r.cloudProviderConfigOutputFile) != 0 {
+		if !(len(r.clusterInfrastructureInputFile) != 0 && len(r.cloudProviderConfigOutputFile) != 0) {
+			return fmt.Errorf("clulster-infrastructure-file and cloud-provider-config-output-file must be specified.")
+		}
+		if err := kubecloudconfig.ValidateFile(r.clusterInfrastructureInputFile); err != nil {
+			return err
+		}
+	}
```

Comment:
what if we do?

```suggestion
	// Validate all files are specified when specifying infrastructure and configmap files
	if len(r.clusterInfrastructureInputFile) !=  len(r.cloudProviderConfigOutputFile) {
		return fmt.Errorf("cluster-infrastructure-file and cloud-provider-config-output-file must be specified.")
	}
	if len(r.clusterInfrastructureInputFile) != 0 {
		kubecloudconfig.ValidateFile(r.clusterInfrastructureInputFile)
	}
```

Reply:
I don't understand what testing if the length of the input file matches the length of the output file accomplishes here since they can be different lengths, so I have left that code intact. I have added the if clause before file validation.